sklearn/feature_selection/_rfe.py

```diff
@@ -155,8 +156,11 @@ def _fit(self, X, y, step_score=None):
         # self.scores_ will not be calculated when calling _fit through fit
 
         tags = self._get_tags()
+        type_y = type_of_target(y)
+        multioutput = 'multioutput' in type_y
         X, y = check_X_y(X, y, "csc", ensure_min_features=2,
-                         force_all_finite=not tags.get('allow_nan', True))
+                         force_all_finite=not tags.get('allow_nan', True),
+                         multi_output=multioutput)
```

Comment:
Why can't we just use multi_output=True in all cases?

Reply:
Yeah! we can do that as well, but check_array function in validation is unnecessarily called! if thats okie I can just change that

Reply:
Yeah! that's actually possible! I thought we may doing extra computation by calling check_array function in validation.py. I will change that now :)